src/scripts/main.js

```diff
@@ -1,5 +1,35 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+wall.addEventListener('click', event => {
+  if (!event.target.closest('.wall') || event.target === spider) {
+    return;
+  }
+
+  moveSomeObject(event, spider, wall);
 });
+
+function moveSomeObject(event, someObject, area) {
+  const maxRight = area.clientWidth - someObject.offsetWidth;
+  const halfObjectWidth = someObject.offsetWidth / 2;
+  const maxDown = area.clientHeight - someObject.offsetHeight;
+  const halfObjectHeight = someObject.offsetHeight / 2;
+
+  if (event.offsetX <= halfObjectWidth) {
+    someObject.style.left = '0px';
+  } else if (event.offsetX >= maxRight + halfObjectWidth) {
+    someObject.style.left = `${maxRight}px`;
+  } else {
+    someObject.style.left = `${event.offsetX - halfObjectWidth}px`;
+  }
+
+  if (event.offsetY <= halfObjectHeight) {
+    someObject.style.top = '0px';
+  } else if (event.offsetY >= maxDown + halfObjectHeight) {
+    someObject.style.top = `${maxDown}px`;
+  } else {
+    someObject.style.top = `${event.offsetY - halfObjectHeight}px`;
+  }
+}
```

Comment:
this function is too complicated and it's not obvious what it is for, so let's refactor!
1) transfer all variables outside the function and pass inside the ones you really need
2) also these 2 if-statements are very similar, try to combine them into one (the function will calculate either X or Y position depending on the arguments passed inside)
3) function shouldn't change style.top or style.left inside of it, it'll be better if it only returns the final value 
``someObject.style.top = `${calcPosition(someArgs)}px`;``
4) also let's rename it, cause after all our changes it won't move anything, but just calculate some value and return it back

Reply:
maxPosition